src/AsciiDocSymbolProvider.ts

```diff
@@ -1,35 +1,77 @@
 import {
     languages,
-    Disposable,
     CancellationToken,
-    Location,
-    Position,
-    TextDocument,
-    SymbolInformation,
-    SymbolKind
+    Disposable,
+    DocumentSymbol,
+    SymbolKind,
+    TextDocument
 } from 'vscode';
 
 export default function registerDocumentSymbolProvider(): Disposable {
-    
-        const _atxPattern = /^(=|#){1,6}\s+.+/;
+
+        const _atxPattern = /^((=|#){1,6})\s+(.+)$/;
+        const _blockPattern = /^[.\-]{4}$/;
 
         return languages.registerDocumentSymbolProvider('asciidoc', {
-    
-            provideDocumentSymbols(document: TextDocument, token: CancellationToken): SymbolInformation[] {
-    
-                const result: SymbolInformation[] = [];
+
+            provideDocumentSymbols(document: TextDocument, token: CancellationToken): DocumentSymbol[] {
+
+                const symbols: DocumentSymbol[] = [];
                 const lineCount = Math.min(document.lineCount, 10000);
+
+                // Really basic way of ignoring literal blocks when parsing headers.
+                let currentBlock: string = null;
+
                 for (let line = 0; line < lineCount; line++) {
-                    const {text} = document.lineAt(line);
-    
-                    if (_atxPattern.test(text)) {
-                        // atx-style, 1-6 = characters
-                        result.push(new SymbolInformation(text, SymbolKind.File, '',
-                            new Location(document.uri, new Position(line, 0))));
+                    const {text, range} = document.lineAt(line);
+
+                    if (_blockPattern.test(text)) {
+                        // Entering a block.
+                        if (currentBlock === null) {
+                            currentBlock = text;
+                        }
+
+                        // Exiting a block.
+                        else if (text === currentBlock) {
+                            currentBlock = null;
+                        }
+
+                        continue;
+                    }
+
+                    // Ignore lines in a literal block.
+                    if (currentBlock !== null) {
+                        continue;
+                    }
+
+                    const match = _atxPattern.exec(text);
+                    if (match !== null) {
+                        const depth = match[1].length;
+                        const symbol = new DocumentSymbol(
+                            match[3],
```

Comment:
I opted to _not_ include the leading syntax of the header in the outline view. This deviates from the behavior in Markdown, but it seems unnecessarily noisy to me.

So instead of showing `== My Section` in the outline we just show `My Section`.